golos.publication/golos.publication.cpp

```diff
@@ -534,8 +545,12 @@ void publication::set_vote(name voter, name author, string permlink, int16_t wei
 
     tables::message_table message_table(_self, author.value);
     auto message_index = message_table.get_index<"bypermlink"_n>();
-    auto mssg_itr = message_index.find(permlink);
+    auto mssg_itr = message_index.find({permlink, ref_block_num});
     eosio_assert(mssg_itr != message_index.end(), "Message doesn't exist.");
+
+    if (is_close_message_by_ref_block_num(ref_block_num))
```

Comment:
For this task, we only change message identification. Checks for actions with posts do not change yet.